src/main/java/core/basesyntax/storage/impl/StorageImpl.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.storage.impl;
+
+import core.basesyntax.storage.Storage;
+import java.util.Collection;
+import java.util.HashMap;
+
+public class StorageImpl implements Storage {
+    private final HashMap<String, StorageEntry> entries;
```

Comment:
The use of 'HashMap' as a reference type should be replaced with 'Map' to follow the best coding practices and to ensure that the code is against the interface rather than a specific implementation.